app/src/main/java/de/xikolo/storages/RecentCoursesStorage.kt

```diff
@@ -0,0 +1,51 @@
+package de.xikolo.storages
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import de.xikolo.storages.base.BaseStorage
+
+typealias RecentCourse = Pair<String, String>
+
+const val MAX_SHORTCUTS = 4
+
+class RecentCoursesStorage : BaseStorage(PREF_RECENT_COURSES, Context.MODE_PRIVATE) {
+
+    companion object {
+        private const val PREF_RECENT_COURSES = "preference_recent_courses"
+
+        fun addCourseToRecentCourses(
+            courseId: String,
+            title: String,
+            courses: LinkedHashSet<RecentCourse>
+        ): LinkedHashSet<RecentCourse> {
+            return courses
+                // filter duplicates
+                .filter { it.first != courseId }
+                .toMutableSet()
+                // add visited course
+                .apply { this.add(Pair(courseId, title)) }
+                // take last MAX_SHORTCUT courses
+                .reversed()
```

Comment:
This reverse is called on a `Set`, is it?

Reply:
No, its applied to a `LinkedHashSet`. But removing the cast to `MutableSet` will break the usage of  `add()` in the apply lambda.

Reply:
I think the tests prove that this is working as expected.